File: dalambert.m
clear, clf, hold off; n=0; h=1;
% Condiciones Iniciales
t=0; x=0; v=1.0; tfin = 8; t=0;
% Inicio de la Simulacion
for t=0:2.0:tfin
    n=0;
    px=[]; py=[]; pz=[];
    for ix=-1:0.01:1
        x = ix+v*t;
        n = n + 1;
        y = (x-v*t);
        px(n)=x;
        py(n)=y;  
    end
     plot(px,py); grid on; xlabel('x'); ylabel('y'); hold on
     pause(0.2)
end

File: dalambert.py
import numpy as np
import matplotlib.pyplot as plt
import time

plt.clf()
plt.grid(True)

# Condiciones Iniciales
v = 1.0
tfin = 8

# Inicio de la simulación
for t in np.arange(0, tfin + 0.1, 2.0):
    px = []
    py = []

    for ix in np.arange(-1, 1.01, 0.01):
        x = ix + v * t
        y = x - v * t
        px.append(x)
        py.append(y)

    plt.plot(px, py)
    plt.xlabel("x")
    plt.ylabel("y")
    plt.grid(True)
    plt.pause(0.2)

plt.show()
